File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.py
class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:
        ans = 0
        turned = list(map(list, zip(*grid))) # turned 90 degree
        for col in turned:
            if col in grid:
                ans += grid.count(col)
       
        return ans

        

File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.cpp
class Solution {
public:
    int equalPairs(vector<vector<int>>& grid) {
        size_t n = grid.size();
        map<vector<int>, int> counts;
        for (auto row:grid) {
            counts[row]++;
        }
        
        int ans = 0;
        for (int i = 0; i < n; i++) {
            vector<int> col = {};
            for (int j = 0; j < n; j++) {
                col.push_back(grid[j][i]);
            }
            ans += counts[col];
        } return ans;
    }
};